Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp

```diff
@@ -104,21 +105,134 @@ std::optional<RTCRtpCapabilities> PeerConnectionBackend::senderCapabilities(Scri
 }
 #endif // USE(LIBWEBRTC) || USE(GSTREAMER_WEBRTC)
 
+#if PLATFORM(WPE) || PLATFORM(GTK)
+class JSONFileHandler {
+public:
+    JSONFileHandler(const String& path)
+        : m_logFile(FilePrintStream::open(path.utf8().data(), "w"))
+    {
+        // Prefer unbuffered output, so that we get a full log upon crash or deadlock.
+        setvbuf(m_logFile->file(), nullptr, _IONBF, 0);
+    }
+
+    void log(String&& event)
+    {
+        m_logFile->println(WTFMove(event));
+    }
+
+    void addClient(uint64_t identifier)
+    {
+        Locker lock(m_clientsLock);
+        m_clients.append(identifier);
+    }
+
+    void removeClient(uint64_t identifier)
+    {
+        Locker lock(m_clientsLock);
+        if (!m_clients.contains(identifier))
+            return;
+
+        m_clients.remove(identifier);
+        if (m_clients.isEmpty())
+            m_logFile = nullptr;
+    }
+
+private:
+    std::unique_ptr<FilePrintStream> m_logFile;
+    Lock m_clientsLock;
+    Vector<uint64_t> m_clients WTF_GUARDED_BY_LOCK(m_clientsLock);
+};
+
+JSONFileHandler& jsonFileHandler(const String& path)
+{
+    static NeverDestroyed<JSONFileHandler> sharedInstance(path);
```

Comment:
What happens if `jsonFileHandler ` is called with different `path` values?
Would it be simpler to not pass the parameter and use `Singleton` as well for aligning with coding style.

Reply:
I'm not sure what you mean, the path comes from an env var, env vars are global within the same process, the same env var can't have different values in that context.

Reply:
`path` argument removed, the getenv is done within the singleton accessor now.